### PR TITLE
Run ONNX cuda backend test cases via ROCm

### PR DESCRIPTION
**Pull request by bddppq**

### CODE REVIEW

**Review by houseroad — APPROVED** [review]
Looks good.

File: caffe2/python/onnx/tests/onnx_backend_test.py [hunk 1]
```diff
@@ -13,6 +13,9 @@
 
 import caffe2.python.onnx.backend as c2
 
+from caffe2.python import core
+core.SetEnginePref({}, {})
```
Comment (houseroad) [inline caffe2/python/onnx/tests/onnx_backend_test.py]: What's the purpose of this code?

Comment (bddppq) [reply caffe2/python/onnx/tests/onnx_backend_test.py]: So that we test the default gpu implementation (instead of cudnn and miopen). Ideally we probably should test both (use cudnn and not).

**Comment by bddppq** [comment]
@petrex @rohithkrn Looks like hip softmax has some flakiness:
```
02:02:28 E           Not equal to tolerance rtol=0.001, atol=1e-07
02:02:28 E           
02:02:28 E           x and y nan location mismatch:
02:02:28 E            x: array([[[0.104313, 0.026669, 0.047564, 0.168046, 0.115689],
02:02:28 E                   [0.047495, 0.046221, 0.020795, 0.019817, 0.026949],
02:02:28 E                   [0.020643, 0.076525, 0.038259, 0.020187, 0.02786 ],...
02:02:28 E            y: array([[[     nan,      nan,      nan,      nan,      nan],
02:02:28 E                   [     nan,      nan,      nan,      nan,      nan],
02:02:28 E                   [     nan,      nan,      nan,      nan,      nan],...
```

The error should be reproducible by running `python caffe2/python/onnx/tests/onnx_backend_test.py OnnxBackendNodeModelTest.test_softmax_axis_1_cuda`

**Comment by rohithkrn** [comment]
@bddppq we will have a look. 
cc @ashishfarmer

**Comment by bddppq** [comment]
Run multiple times to test flakiness:

https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-test/5280/
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-devtoolset7-rocmrpm-centos7.5-test/2698/

https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-test/5281/
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-devtoolset7-rocmrpm-centos7.5-test/2699/

https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-test/5282/
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-devtoolset7-rocmrpm-centos7.5-test/2700/
